src/GafferSceneUI/Inspector.cpp

```diff
@@ -91,29 +92,75 @@ Gaffer::Plug *sourceInput( Gaffer::Plug *plug )
 /// return an output plug, such as when it is connected to an anim curve. This
 /// seems to suggest that maybe there are two concepts? Something like a pure
 /// `contextAwareSource()` and a `userEditableSource()`?
-Gaffer::Plug *spreadsheetAwareSource( Gaffer::Plug *plug )
+
+Gaffer::Plug *contextAwareSource( Gaffer::Plug *plug )
 {
 	if( auto sourceValuePlug = plug->source<ValuePlug>() )
 	{
 		if( auto spreadsheet = runTimeCast<Spreadsheet>( sourceValuePlug->node() ) )
 		{
 			if( spreadsheet->outPlug()->isAncestorOf( sourceValuePlug ) )
 			{
-				Gaffer::Plug *inPlug = spreadsheet->activeInPlug( sourceValuePlug );
-				if( inPlug->ancestor<Spreadsheet::RowPlug>() == spreadsheet->rowsPlug()->defaultRow() )
-				{
-					// Don't want to edit the default row, as that could affect
-					// all sorts of other things.
-					return nullptr;
-				}
-				return sourceInput( inPlug );
+				return sourceInput( spreadsheet->activeInPlug( sourceValuePlug ) );
 			}
 		}
 	}
 
 	return sourceInput( plug );
 }
 
+Gaffer::Plug *userEditableSource( Gaffer::Plug *plug )
+{
+	Gaffer::Plug *p = contextAwareSource( plug );
+
+	if( auto spreadsheet = runTimeCast<Spreadsheet>( p->node() ) )
+	{
+		if( p->ancestor<Spreadsheet::RowPlug>() == spreadsheet->rowsPlug()->defaultRow() )
+		{
+			// Don't want to edit the default row, as that could affect
+			// all sorts of other things.
+			return nullptr;
+		}
+	}
+
+	return p;
+}
+
+std::string lockedReason( const Plug *plug, const EditScope *editScope )
```

Comment:
This also feels like it might be redundant, and could be absorbed into `nonEditableReason()`.